client/azure.go

```diff
@@ -123,83 +150,81 @@ func ResponseWasNotFound(resp autorest.Response) bool {
 	return false
 }
 
-func newDeploymentsClient(config AzureConfig) (*resources.DeploymentsClient, error) {
-	spt, err := newServicePrincipalToken(config, azure.PublicCloud.ResourceManagerEndpoint)
-	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
-	}
-
-	client := resources.NewDeploymentsClient(config.SubscriptionID)
-	client.Authorizer = autorest.NewBearerAuthorizer(spt)
+func newDeploymentsClient(config *azureClientConfig) (*resources.DeploymentsClient, error) {
+	client := resources.NewDeploymentsClientWithBaseURI(
+		config.resourceManagerEndpoint,
```

Comment:
We need to use `config.resourceManagerEndpoint` of specific cloud. By default `NewDeploymentsClient` uses public one.